### PR TITLE
PBI 1 register

### PR DESCRIPTION
**Pull request by Zsaschz**
### Description
Impelement user controller and register

### Related Issues
#4

### CODE REVIEW

File: src/Controllers/LMSController.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Uasoft\Badaso\Module\LMS\Controllers;
+
+use Uasoft\Badaso\Controllers\Controller;
+
+class LMSController extends Controller
+{
+}
```
Comment (nofamex) [inline src/Controllers/LMSController.php]: what is this piece of code

File: tests/Feature/BadasoLMSRegisterTest.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+use Illuminate\Support\Str;
+use Tests\TestCase;
+use Uasoft\Badaso\Helpers\CallHelperTest;
+use Uasoft\Badaso\Module\LMSModule\Models\User;
+
+class BadasoLMSRegisterTest extends TestCase
+{
+    public function testStartInit()
+    {
+        // init user login
```
Comment (nofamex) [inline tests/Feature/BadasoLMSRegisterTest.php]: remove comment

File: tests/Feature/BadasoLMSRegisterTest.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        CallHelperTest::handleUserAdminAuthorize($this);
+    }
+
+    /** @test */
```
Comment (nofamex) [inline tests/Feature/BadasoLMSRegisterTest.php]: remove comment

**Review by nofamex — APPROVED** [review]

File: src/Controllers/AuthController.php [hunk 4]
```diff
@@ -5,13 +5,55 @@
...
+                'username' => 'required|string|max:255',
+                'password' => 'required|string|max:255',
+                'password_confirmation' => 'required|string|max:255',
+            ]);
+
+            if ($request->input('password') != $request->input('password_confirmation')) {
+                return ApiResponse::failed('Password does not match, please try again');
+            }
+
+            $user = User::factory()->create([
```
Comment (AldiNFitrah) [inline src/Controllers/AuthController.php]: factory supposed to be used only in tests. Please just use `->create` instead

File: tests/Feature/RegisterTest.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            'email' => $name.'@gmail.com',
+            'password' => Hash::make($name),
+        ];
+        $user = User::create($create_user);
+        $this->assertEquals($create_user['name'], $user->name);
+        $this->assertEquals($create_user['username'], $user->username);
+        $this->assertEquals($create_user['email'], $user->email);
+        $this->assertEquals($create_user['password'], $user->password);
+
+        $user->delete();
```
Comment (AldiNFitrah) [inline tests/Feature/RegisterTest.php]: no need to delete it. Don't worry about the database since it's always getting refreshed each time we run a test.

**Review by AldiNFitrah — APPROVED** [review]
LGTM, just those 2 minor issues.

**Pull request merged**